dask/array/core.py

```diff
@@ -2702,65 +2711,37 @@ def insert_to_ooc(arr, out, lock=True, region=None, return_stored=False):
         lock = Lock()
 
     slices = slices_from_chunks(arr.chunks)
+    if region:
+        slices = [fuse_slice(region, slc) for slc in slices]
 
     name = 'store-%s' % arr.name
+    sto_chunk = store_chunk
```

Comment:
I'm not sure I understand this aliasing or the purpose behind the name `sto_chunk`

Reply:
Basically we are trying to handle two cases:

1. `store` called with `compute=False, return_stored=True`
    (a.k.a. `return_stored=True, load_stored=True` in this context)
2. `store` called with anything else

The first case means that we want to store the results and reload them, but don't want to start computation quite yet. In this case, we are better served by acquiring the lock once to perform both the store and load steps, which is what we try to do. This is what the content under that `if` is setting up. 

In the second case, we either want storing to happen asynchronously of loading or we don't want to load the data at all. Thus we deal with the store step as an independent step that may or may not be chained to a load step. Both steps handling their own locking. This is what comes before the `if`.

Admittedly we could probably override what `store_chunk` means in `insert_to_ooc` instead of assigning to `sto_chunk`. Was a little reluctant to do this personally, but could be persuaded otherwise.

Reply:
In cases like this in the past we've just used a `chunk` or `func` variable name. `sto_chunk` looks like a typo to me. No strong feelings though.

Reply:
I picked `sto` because it is sometimes used as an abbreviation of `store` (e.g. the assembly instruction). Though if `func` were preferred, could do that instead.